### ISSUE TITLE
Multiple episode thumbnails for merged eps

### ISSUE THREAD

**Issue by GoogleCodeExporter** [TV-Rename/tvrename#44]
```
Do something nice for the thumbnails of merged episodes
 * In episode guide
 * On disk for .tbn files?
```

Original issue reported on code.google.com by `tvren...@tvrename.com` on 10 Sep 2009 at 12:57

**Comment by MarkSummerville** [comment]
https://kodi.wiki/view/TV-Show_artwork#Episode_Thumbnail does not mention what to do for multi episodes

**Comment by JoeBiellik** [comment]
I've not seen any media centre software that supports multiple episode thumbs, merged ep or not.

**Comment by Andy-ABTec** [comment]
KODI!

**Comment by JoeBiellik** [comment]
That's very interesting, I'd like to know if it got the thumbs from TVDB or just picked the frame from the video. I'm guessing its from the video file otherwise what would it display if you split a double ep into three or more parts?

**Comment by Andy-ABTec** [comment]
The TVDB has two episodes listed and the thumbs came from there...
Messing around with "The Gifted" (ep 12 and 13 are a double) you need to manually set the episode bookmark at the correct point in time, it would appear that Kodi initially sets the artwork for both episodes to be the same image from the first episode, but on completing a scan for new content on the show Kodi correctly sets both episode thumbnails from The TVDB.

**Comment by Andy-ABTec** [comment]
It would appear that there is a table in the Kodi database specifically for this stuff

![image](https://user-images.githubusercontent.com/5175695/36060258-dd71ddfc-0e3d-11e8-8271-231b3c3851d9.png)

**Comment by JoeBiellik** [comment]
Thanks for doing the research, it looks like this is something we should support.

**Comment by Andy-ABTec** [comment]
No problem...

**Comment by MarkSummerville** [comment]
Great Research - Any ideas on how we can set this stuff via NFO files and naming of images (which is all we control at the moment)

**Comment by Andy-ABTec** [comment]
I don't think there's much that can be done. Having created a list of double episodes from my library and checking on thetvdb it would seem that thetvdb always processes single episodes which seems quite logical, as its easy to make a double episode out of two singles but not so easy to go the other way.

I think a lot of the "double episode" stuff is created by the broadcasters not the show producers I cant cite specifics at the moment but I know that some double episodes I've downloaded turn out to be two single episodes on the DVD when it is released.

Taking Agents of S.H.I.E.L.D. Season 5 episodes 1 and 2 as an example the head of thetvdb data for each show looks like this:
```
{                                                 {
  "data": {                                         "data":{
    "id": 6276702,                                    "id": 6407853,
    "airedSeason": 5,                                 "airedSeason": 5,
    "airedSeasonID": 726439,                          "airedSeasonID": 726439,
    "airedEpisodeNumber": 1,                          "airedEpisodeNumber": 2,
    "episodeName": "Orientation (1)",                 "episodeName": "Orientation (2)",
    "firstAired": "2017-12-01",                       "firstAired": "2017-12-01",
    "guestStars": [                                   "guestStars": [
      "Jeff Ward",                                      "Jeff Ward",
``` 
The only hint that this was initially broadcast (and downloaded) as a double episode is that both episodes have the same `"firstAired" date. But obviously thats not to say it will be broadcast that way again.

I'll go and have a rummage in my DVD collection and see if I can find a genuine double episode and then check how thetvdb represents it...

**Comment by Andy-ABTec** [comment]
OK, from what I can tell Defiance Season 3 Episodes 1 and 2 is a genuine double episode, it is 120" long and watching (the DVD) from 38" to 42" there is no obvious splice point. it's almost a continuous run of video with no fades to black or other artifacts.

Looking at thetvdb content for this I see two episodes with the same air date:

![image](https://user-images.githubusercontent.com/5175695/36071947-fff983ee-0f0e-11e8-94c2-4a37002b9cfb.png)

From this I can only assume that thetvdb treats everything as a single episodes. I had to merge episodes 1 and 2 in TV Rename to stop episode 2 being flagged as missing.

To sum up: I think the only chance you have of processing this is to look for duplicate  `firstAired` dates
if the downloaded show filename contains something like S03E01-E02 or similar but it's flakey at best

**Comment by MarkSummerville** [comment]
We know in TVR that the user has elected to merge the episode. We then provide the episode with a different name and an updated NFO file to indicate to Kodi that it is a merged episode. Any idea how we can provide jpg files in such a way that Kodi recognises them as the two individual episodes?

**Comment by JoeBiellik** [comment]
I added Agents of S.H.I.E.L.D. season 5 to Kodi, had it scrape TVDB and then exported the media library which only returned one thumb "Marvel's Agents of S.H.I.E.L.D. - S05E01-E02 - Orientation-thumb.jpg", so I'm not sure how we would go about providing it with two.

**Comment by Andy-ABTec** [comment]
See the attached .nfo files downloaded from thetvdb:

[orientation-1.txt](https://github.com/TV-Rename/tvrename/files/1714521/orientation-1.txt)
[orientation-2.txt](https://github.com/TV-Rename/tvrename/files/1714522/orientation-2.txt)

Is that not a relative pathname at the bottom of each file? Look like the episode images to me!

https://www.thetvdb.com/banners/episodes/263365/6276702.jpg

https://www.thetvdb.com/banners/episodes/263365/6407853.jpg

**Comment by JoeBiellik** [comment]
@Andy-ABTec The issue here is not that we can't get the images, but that we don't know where to put them to make Kodi index them from disk.

**Comment by Andy-ABTec** [comment]
Mmm.... That's weird - I've just exported my video library again and according to the file structure you are absolutely correct s05e02 does not exist!
Actually looking at the show in Kodi I see two thumbnails but they appear to share one nfo file. All I got in my export was the season 05 episode 01 files so the image for season 05 episode 02 only exists within Kodi itself. its not exported. 
I think all you can do is call the unnamed image according to Kodi thumbnail rules and leave it in the relevant folder, at least that way its available if people wish to browse to it from within Kodi.

**Comment by MarkSummerville** [comment]
OK. I think I have got to the bottom of what we can do:

For example S14E01-E02 is a joint episode.

1. Store individual episode thumbs in the directory S14E01-thumb.jpg and S14E02-thumb.jpg
2. Update the rename check so that it knows not to rename either of these to S14E01-E02 (which it will do at the moment)
3. Explicitly add the `<thumb>` and maybe the `<art>` tags in the NFO file for each `episodedetails`. The correct one for each `episodedetails`
4. include a block as follows in all non-first `episodedetails`:
```
           <resume>
               <position>0.000000</position>
               <total>0.000000</total>
           </resume>
```

I can do all this, but wouldn't mind someone checking manually that this works for them first

**Comment by Andy-ABTec** [comment]
Happy to check for you... Can you give me a bit more detail about 3 and 4 please - a tweaked nfo file would be good...

**Comment by MarkSummerville** [comment]
Maybe don't worry. I can;t make it work myself anymore. Even if I export one of heir files and refresh that, I can't get to to use different thumb for a multipart episode without setting it manually. I got it to work this morning, but not this afternoon

**Comment by MarkSummerville** [comment]
Raised as https://trac.kodi.tv/ticket/17771

**Comment by MarkSummerville** [comment]
Finally got it fixed!
